core/src/commonMain/kotlin/Throwable.kt

```diff
@@ -0,0 +1,13 @@
+package com.juul.kable
+
+import kotlinx.coroutines.CancellationException
+
+internal fun Throwable.unwrap(): Throwable {
```

Comment:
This function is super confusing. I think I'd rather something like `Throwable.unwrapCancellationCause(): Throwable?` but even that doesn't really capture the semantics. Can we get a doc comment explaining the intent of what this unwraps into and when?